### PR TITLE
Fix npm run watch crashes when updates are made while a world is running

### PR DESCRIPTION
**Pull request by CarlosFdez**
static/packs will need to be deleted from people's machines for builds to work properly after this. Perhaps I should delete it just in case?

After this static files are not deleted (only overwritten) between refreshes in npm run watch, but they are deleted when `npm run watch` or `npm run build` is ran. Perhaps that is fine.

### CODE REVIEW

**Comment by stwlam** [comment]
Yeah could we just delete the static/packs directory and recreate it every time? It should be possible to do it blindly without throwing an error if nothing is there

**Comment by CarlosFdez** [comment]
The problem is that vite doesn't seem configurable enough to selectively copy only *some* static files.

1) Vite currently wipes everything in dist everytime it rebuilds. This explodes due to file locks. 
2) Vite copies everything in static over to dist on every rebuild. If one of those folders is "packs", this explodes due to file locks.

**Comment by CarlosFdez** [comment]
I guess as a reminder, this is the current vite build flow you set up

1) build packs writes packs to static/packs
2) vite runs, doing these subordinate steps on the first and every rebuild
    1) wipe dist (due to emptyOutDir)
    2) copy everything in static/ to dist/  (due publicDir)
    3) Do the rest of the build this part doesn't matter to foundry

To avoid the crash due to wiping dist/: we need to make sure dist/ is cleaned only on the first run. That's npm run clean. 
To avoid the crash due to overwriting pack files: packs must not exist in static/ during rebuilds.

In other words, I don't see a need to avoid a separate pre-build step, but if we can configure a pre initial build and a post build step within vite itself, we can do the clean there and delete static/packs after the first run. Is this what you were getting at?

**Comment by CarlosFdez** [comment]
For npm run hot at least, it seems that the vite config is reloaded when you change it. I think attempting to do the clean there is going to end up being a bit bug prone, especially if we make packs not go into dist/ after the first build somehow.
